jax/_src/numpy/lax_numpy.py

```diff
@@ -749,9 +749,20 @@ def _compute_newshape(a, newshape):
   except: iterable = False
   else: iterable = True
   newshape = core.canonicalize_shape(newshape if iterable else [newshape])
-  return tuple(- core.divide_shape_sizes(np.shape(a), newshape)
-               if core.symbolic_equal_dim(d, -1) else d
-               for d in newshape)
+  # One dimension in newshape can be -1
+  newshape_has_m1 = False
```

Comment:
This code can be simplified if we decide that we don't support `-1` dimensions in `reshape`. The user typically has a reason to believe that the reshape is correct, and hence it could write the actual missing dimension.